self_enrollment_tool/views.py

```diff
@@ -0,0 +1,267 @@
+                                 f'SIS ID ({cc["sis_course_id"]}) for Canvas course {ci.canvas_course_id}. Aborting.')
+                    messages.error(request, f'Course instance ID ({course_search_term}) does not match Canvas '
+                                            f'course SIS ID ({cc["sis_course_id"]}) for Canvas course '
+                                            f'{ci.canvas_course_id}. Aborting.')
+                    context['abort'] = True
+
+                # Check if it is an ILE or SB course. (source != xmlfeed)
+                if ci.source == 'xmlfeed':
+                    logger.warning(f'Course instance ID ({course_search_term}) is not an ILE course. Aborting.')
+                    messages.error(request, f'Course instance ID ({course_search_term}) is not an ILE/SB course.')
+                    context['abort'] = True
+
+            else:
+                logger.error(f'Course instance {ci.course_instance_id} does not have a Canvas course ID set.')
+                messages.error(request, f'Course instance {ci.course_instance_id} does not have a Canvas course ID set. Cannot continue.')
+                context['abort'] = True
+
+        except CourseInstance.DoesNotExist:
+            logger.exception('Could not determine the course instance for Canvas '
+                             'course instance id %s' % course_search_term)
+            messages.error(request, 'Could not find a Course Instance from search term')
+            context['abort'] = True
+        except CanvasAPIError:
+            logger.exception(f'Could not find Canvas course {course_search_term} via Canvas API')
+            messages.error(request, f'Could not find Canvas course {course_search_term}. Aborting.')
+            context['abort'] = True
+    else:
+        logger.warning(f'course search term {course_search_term} not numeric.')
+        messages.error(request, 'Search term must be populated and may only be numbers')
+        context['abort'] = True
+
+    # fetch the roles for the dropdown
+    roles = get_canvas_roles()
+    if roles:
+        context['roles'] = roles
+    else:
+        logger.error(f'Unable to retrieve roles for Canvas course {course_search_term}. Response from get_canvas_roles={roles}')
+        messages.error(request, f'Error fetching roles for SIS ID {course_search_term}. Aborting.')
+        context['abort'] = True
+
+    return render(request, 'self_enrollment_tool/add_new.html', context)
+
+
+@login_required
+@lti_role_required(const.ADMINISTRATOR)
+@lti_permission_required(settings.PERMISSION_SELF_ENROLLMENT_TOOL)
+@require_http_methods(['GET'])
+def add_new(request):
+    context = {}
+    return render(request, 'self_enrollment_tool/add_new.html', context)
+
+
+@login_required
+@lti_role_required(const.ADMINISTRATOR)
+@lti_permission_required(settings.PERMISSION_SELF_ENROLLMENT_TOOL)
+@require_http_methods(['POST'])
+def enable(request, course_instance_id):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function calls 17 other functions. By comparison, 98% of the functions in the CodeGuru reference dataset use fewer. This indicates the function is highly coupled with other functions. A function that is highly coupled with other functions is difficult to understand and its behavior might change unexpectedly when one of its referenced functions is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this function or break it into multiple functions.